src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -67,26 +70,50 @@
     @Mock
     private GdprService gdprService;
 
-    private Clock clock;
     private Timeout timeout;
     private PrivacyEnforcementService privacyEnforcementService;
 
     @Before
     public void setUp() {
         given(bidderCatalog.bidderInfoByName(anyString())).willReturn(givenBidderInfo(15, true));
 
-        clock = Clock.fixed(Instant.now(), ZoneId.systemDefault());
-        timeout = new TimeoutFactory(clock).create(500);
+        given(gdprService.isGdprEnforced(any(), any(), any(), any()))
+                .willReturn(Future.succeededFuture(true));
+        given(gdprService.resultByVendor(any(), any(), any(), any(), any()))
+                .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, false), null)));
+
+        timeout = new TimeoutFactory(Clock.fixed(Instant.now(), ZoneId.systemDefault())).create(500);
 
         privacyEnforcementService = new PrivacyEnforcementService(gdprService, bidderCatalog, metrics, false);
     }
 
+    @Test
+    public void shouldTolerateEmptyBidderToUserMap() {
+        // given and when
+        final BidRequest bidRequest  = givenBidRequest(emptyList(),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(null)
+                        .device(null)
+                        .regs(null));
+
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(emptyMap(), null, null, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        verify(gdprService).isGdprEnforced(isNull(), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verify(gdprService).resultByVendor(eq(singleton(15)), isNull(), any(), any(), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).isEqualTo(emptyMap());
+    }
+
     @Test
     public void shouldNotMaskWhenDeviceLmtIsNullAndExtRegsGdprIsOneAndGdprServiceRespondIsGdprEnforcedWithTrueAndResultByVendorWithoutEnforcement() {
         // given
-        given(gdprService.isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+        given(gdprService.isGdprEnforced(any(), any(), any(), any()))
                 .willReturn(Future.succeededFuture(true));
-        given(gdprService.resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout)))
+        given(gdprService.resultByVendor(any(), any(), any(), any(), any()))
                 .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, true), null)));
```

Comment:
I guess you've forgot to remove these two stubs (`isGdprEnforced` and `resultByVendor`) since you already have them in your `@Before setUp` method.